src/python_testing/TC_AVSM_2_1.py

```diff
@@ -264,20 +284,23 @@ async def test_TC_AVSM_2_1(self):
                 endpoint=endpoint, cluster=cluster, attribute=attr.AllocatedVideoStreams
             )
             logger.info(f"Rx'd AllocatedVideoStreams: {value}")
+            asserts.assert_is_not_none(value, "AllocatedVideoStreams is None")
 
         self.step(18)
         if self.pics_guard(self.check_pics("AVSM.S.A0010")):
             value = await self.read_single_attribute_check_success(
                 endpoint=endpoint, cluster=cluster, attribute=attr.AllocatedAudioStreams
             )
             logger.info(f"Rx'd AllocatedAudioStreams: {value}")
+            asserts.assert_is_not_none(value, "AllocatedAudioStreams is None")
 
         self.step(19)
         if self.pics_guard(self.check_pics("AVSM.S.A0011")):
             value = await self.read_single_attribute_check_success(
                 endpoint=endpoint, cluster=cluster, attribute=attr.AllocatedSnapshotStreams
             )
             logger.info(f"Rx'd AllocatedSnapshotStreams: {value}")
+            asserts.assert_is_not_none(value, "AllocatedSnapshotStreams is None")
```

Comment:
This also needs to verify the values within the fields of the list of structs

Reply:
Thank you for review. Since snapshot stream is not allocated, the returned list in empty, how should we verify the struct?

Reply:
The contents of AllocatedSnapshotStreams should be tested by a different test case that is functional. Checking that read succeeds here is actually not relevant and is already done by TC-IDM-10.2. If not checking contents, we should not blind-read.

Reply:
What should happen is that the attribute is read, and if not an empty list (as no streams are maybe allocated), then the contents of the struct verified.  I'm fine with doing this in a follow-up PR though.